### PR TITLE
Add Snap Point support to Scrollviewer

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Adds snap behavior to scrollview and scroll content presenter. Panels and item controls can implement `IScrollSnapPointInfo` to report the nature of the contents' snap points.
The following interfaces were added;
```csharp
    public interface IScrollSnapPointsInfo
    {
        bool AreHorizontalSnapPointsRegular { get; }
        bool AreVerticalSnapPointsRegular { get; }

        IReadOnlyList<double> GetIrregularSnapPoints(Orientation orientation, SnapPointsAlignment snapPointsAlignment);
        double GetRegularSnapPoints(Orientation orientation, SnapPointsAlignment snapPointsAlignment, out double offset);

        event EventHandler<RoutedEventArgs> HorizontalSnapPointsChanged;
        event EventHandler<RoutedEventArgs> VerticalSnapPointsChanged;
    }

    public enum SnapPointsAlignment
    {
        Near,
        Center,
        Far
    }
    
    public enum SnapPointsType
    {
        None,
        Optional,
        Mandatory,
        OptionalSingle,
        MandatorySingle
    }
```

ScrollViewer also has been updated with the following properties;
```csharp
        public SnapPointsType HorizontalSnapPointsType { get; set; }
        public SnapPointsType VerticalSnapPointsType { get; set; }
        public SnapPointsAlignment HorizontalSnapPointsAlignment { get; set; }
        public SnapPointsAlignment VerticalSnapPointsAlignment { get; set; }
        public SnapPointsType VerticalSnapPointsType { get; set; }
```

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by grokys** [comment]
This PR is a little hard to test because:

1. I'm not familiar with this feature in UWP
2. It seems to only work with touch (and I don't have a touch device on my main dev machine)
3. There's no example of its usage in e.g. ControlCatalog for me to quickly try
4. There's no unit tests

I'll try to take a look at it at some point, but if you could at least add some unit tests that would be useful.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028719-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028733-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls/Primitives/SnapPointsAlignment.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Avalonia.Controls.Primitives
+{
+    public enum SnapPointsAlignment
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Primitives/SnapPointsAlignment.cs:3]: Can you add documentation for these new types? IScrollSnapPointsInfo, IScrollSnapPointsInfo, IScrollSnapPointsInfo.

Other than that, looks good to me.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028762-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028954-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028993-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028996-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029040-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**